modal/_runtime/container_io_manager.py

```diff
@@ -449,14 +449,17 @@ async def _dynamic_concurrency_loop(self):
 
             await asyncio.sleep(DYNAMIC_CONCURRENCY_INTERVAL_SECS)
 
-    async def get_app_objects(self) -> RunningApp:
-        req = api_pb2.AppGetLayoutRequest(app_id=self.app_id)
-        resp = await retry_transient_errors(self._client.stub.AppGetLayout, req)
-        logger.debug(f"AppGetLayout received {len(resp.app_layout.objects)} objects for app {self.app_id}")
+    async def get_app_objects(self, app_layout: api_pb2.AppLayout) -> RunningApp:
+        if len(app_layout.objects) == 0:
```

Comment:
I assume it's impossible for apps to have zero objects and still end up running a function? :D